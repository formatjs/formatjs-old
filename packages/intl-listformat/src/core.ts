import {
  toObject,
  getOption,
  ListPatternLocaleData,
  unpackData,
  setInternalSlot,
  getCanonicalLocales,
  supportedLocales,
  createResolveLocale,
  getInternalSlot,
  ListPatternFieldsData,
  ListPatternData,
  partitionPattern,
  invariant,
  isLiteralPart,
  LiteralPart,
} from '@formatjs/intl-utils';

export interface IntlListFormatOptions {
  /**
   * The locale matching algorithm to use.
   * Possible values are "lookup" and "best fit"; the default is "best fit".
   * For information about this option, see
   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_negotiation.
   */
  localeMatcher?: 'best fit' | 'lookup';
  /**
   * The format of output message. Possible values are:
   * - "always" (default, e.g., 1 day ago),
   * - or "auto" (e.g., yesterday).
   * The "auto" value allows to not always have to
   * use numeric values in the output.
   */
  type?: 'conjunction' | 'disjunction' | 'unit';
  /**
   * The length of the internationalized message. Possible values are:
   * - "long" (default, e.g., in 1 month)
   * - "short" (e.g., in 1 mo.),
   * - or "narrow" (e.g., in 1 mo.).
   * The narrow style could be similar to the short style for some locales.
   */
  style?: 'long' | 'short' | 'narrow';
}

export interface ResolvedIntlListFormatOptions {
  /**
   * The BCP 47 language tag for the locale actually used.
   * If any Unicode extension values were requested in the
   * input BCP 47 language tag that led to this locale,
   * the key-value pairs that were requested and are
   * supported for this locale are included in locale.
   */
  locale: string;
  /**
   * The format of output message. Possible values are:
   * - "always" (default, e.g., 1 day ago),
   * - or "auto" (e.g., yesterday).
   * The "auto" value allows to not always have to
   * use numeric values in the output.
   */
  type: 'conjunction' | 'disjunction' | 'unit';
  /**
   * The length of the internationalized message. Possible values are:
   * - "long" (default, e.g., in 1 month)
   * - "short" (e.g., in 1 mo.),
   * - or "narrow" (e.g., in 1 mo.).
   * The narrow style could be similar to the short style for some locales.
   */
  style: 'long' | 'short' | 'narrow';
}

export type Part = LiteralPart | ElementPart;

export interface ElementPart {
  type: 'element';
  value: string;
}

interface Placeable {
  type: string;
  value: string;
}

interface ListFormatInternal {
  style: IntlListFormatOptions['style'];
  type: IntlListFormatOptions['type'];
  locale: string;
  templatePair: string;
  templateStart: string;
  templateEnd: string;
  templateMiddle: string;
  initializedListFormat: boolean;
}

function validateInstance(instance: any, method: string) {
  if (!(instance instanceof ListFormat)) {
    throw new TypeError(
      `Method Intl.ListFormat.prototype.${method} called on incompatible receiver ${String(
        instance
      )}`
    );
  }
}

/**
 * https://tc39.es/proposal-intl-list-format/#sec-createstringlistfromiterable
 * @param list list
 */
function stringListFromIterable(list: any[]): string[] {
  if (list === undefined) {
    return [];
  }
  const result = [];
  for (const el of list) {
    if (typeof el !== 'string') {
      throw new TypeError(`array list[${list.indexOf(el)}] is not type String`);
    }
    result.push(el);
  }
  return result;
}

function createPartsFromList(
  internalSlotMap: WeakMap<ListFormat, ListFormatInternal>,
  lf: ListFormat,
  list: string[]
) {
  const size = list.length;
  if (size === 0) {
    return [];
  }
  if (size === 2) {
    const pattern = getInternalSlot(internalSlotMap, lf, 'templatePair');
    const first = {type: 'element', value: list[0]};
    const second = {type: 'element', value: list[1]};
    return deconstructPattern(pattern, {'0': first, '1': second});
  }
  const last = {
    type: 'element',
    value: list[size - 1],
  };
  let parts: Placeable[] | Placeable = last;
  let i = size - 2;
  while (i >= 0) {
    let pattern;
    if (i === 0) {
      pattern = getInternalSlot(internalSlotMap, lf, 'templateStart');
    } else if (i < size - 2) {
      pattern = getInternalSlot(internalSlotMap, lf, 'templateMiddle');
    } else {
      pattern = getInternalSlot(internalSlotMap, lf, 'templateEnd');
    }
    const head = {type: 'element', value: list[i]};
    parts = deconstructPattern(pattern, {'0': head, '1': parts});
    i--;
  }
  return parts;
}

function deconstructPattern(
  pattern: string,
  placeables: Record<string, Placeable | Placeable[]>
) {
  const patternParts = partitionPattern(pattern);
  const result: Placeable[] = [];
  for (const patternPart of patternParts) {
    const {type: part} = patternPart;
    if (isLiteralPart(patternPart)) {
      result.push({
        type: 'literal',
        value: patternPart.value,
      });
    } else {
      invariant(part in placeables, `${part} is missing from placables`);
      const subst = placeables[part];
      if (Array.isArray(subst)) {
        result.push(...subst);
      } else {
        result.push(subst);
      }
    }
  }
  return result;
}

export default class ListFormat {
  constructor(locales?: string | string[], options?: IntlListFormatOptions) {
    // test262/test/intl402/ListFormat/constructor/constructor/newtarget-undefined.js
    // Cannot use `new.target` bc of IE11 & TS transpiles it to something else
    const newTarget =
      this && this instanceof ListFormat ? this.constructor : void 0;
    if (!newTarget) {
      throw new TypeError("Intl.ListFormat must be called with 'new'");
    }
    setInternalSlot(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      'initializedListFormat',
      true
    );
    const requestedLocales = getCanonicalLocales(locales);
    const opt: any = Object.create(null);
    const opts =
      options === undefined ? Object.create(null) : toObject(options);
    const matcher = getOption(
      opts,
      'localeMatcher',
      'string',
      ['best fit', 'lookup'],
      'best fit'
    );
    opt.localeMatcher = matcher;
    const {localeData} = ListFormat;
    const r = createResolveLocale(ListFormat.getDefaultLocale)(
      ListFormat.availableLocales,
      requestedLocales,
      opt,
      ListFormat.relevantExtensionKeys,
      localeData
    );
    setInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'locale', r.locale);
    const type: keyof ListPatternFieldsData = getOption(
      opts,
      'type',
      'string',
      ['conjunction', 'disjunction', 'unit'],
      'conjunction'
    );
    setInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'type', type);
    const style: keyof ListPatternData = getOption(
      opts,
      'style',
      'string',
      ['long', 'short', 'narrow'],
      'long'
    );
    setInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'style', style);
    const {dataLocale} = r;
    const dataLocaleData = localeData[dataLocale];
    const dataLocaleTypes = dataLocaleData![type];
    const templates = dataLocaleTypes![style]!;
    setInternalSlot(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      'templatePair',
      templates.pair
    );
    setInternalSlot(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      'templateStart',
      templates.start
    );
    setInternalSlot(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      'templateMiddle',
      templates.middle
    );
    setInternalSlot(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      'templateEnd',
      templates.end
    );
  }
  format(elements: string[]): string {
    validateInstance(this, 'format');
    let result = '';
    const parts = createPartsFromList(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      stringListFromIterable(elements)
    );
    if (!Array.isArray(parts)) {
      return parts.value;
    }
    for (const p of parts) {
      result += p.value;
    }
    return result;
  }
  formatToParts(elements: string[]): Part[] {
    validateInstance(this, 'format');
    const parts = createPartsFromList(
      ListFormat.__INTERNAL_SLOT_MAP__,
      this,
      stringListFromIterable(elements)
    );
    if (!Array.isArray(parts)) {
      return [parts as Part];
    }
    const result: Part[] = [];
    for (const part of parts) {
      result.push({...part} as Part);
    }
    return result;
  }

  resolvedOptions(): ResolvedIntlListFormatOptions {
    validateInstance(this, 'resolvedOptions');
    return {
      locale: getInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'locale'),
      type: getInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'type')!,
      style: getInternalSlot(ListFormat.__INTERNAL_SLOT_MAP__, this, 'style')!,
    };
  }

  public static supportedLocalesOf(
    locales: string | string[],
    options?: Pick<IntlListFormatOptions, 'localeMatcher'>
  ) {
    // test262/test/intl402/ListFormat/constructor/supportedLocalesOf/result-type.js
    return supportedLocales(
      ListFormat.availableLocales,
      getCanonicalLocales(locales),
      options
    );
  }

  public static __addLocaleData(...data: ListPatternLocaleData[]) {
    for (const datum of data) {
      const availableLocales: string[] = Object.keys(
        [
          ...datum.availableLocales,
          ...Object.keys(datum.aliases),
          ...Object.keys(datum.parentLocales),
        ].reduce((all: Record<string, true>, k) => {
          all[k] = true;
          return all;
        }, {})
      );
      availableLocales.forEach(locale => {
        try {
          ListFormat.localeData[locale] = unpackData(locale, datum);
        } catch (e) {
          // If we can't unpack this data, ignore the locale
        }
      });
    }
    ListFormat.availableLocales = Object.keys(ListFormat.localeData);
    if (!ListFormat.__defaultLocale) {
      ListFormat.__defaultLocale = ListFormat.availableLocales[0];
    }
  }
  static localeData: Record<string, ListPatternFieldsData> = {};
  private static availableLocales: string[] = [];
  private static __defaultLocale = 'en';
  private static getDefaultLocale() {
    return ListFormat.__defaultLocale;
  }
  private static relevantExtensionKeys = [];
  public static polyfilled = true;
  private static readonly __INTERNAL_SLOT_MAP__ = new WeakMap<
    ListFormat,
    ListFormatInternal
  >();
}

try {
  // IE11 does not have Symbol
  if (typeof Symbol !== 'undefined') {
    Object.defineProperty(ListFormat.prototype, Symbol.toStringTag, {
      value: 'Intl.ListFormat',
      writable: false,
      enumerable: false,
      configurable: true,
    });
  }

  // https://github.com/tc39/test262/blob/master/test/intl402/ListFormat/constructor/length.js
  Object.defineProperty(ListFormat.prototype.constructor, 'length', {
    value: 0,
    writable: false,
    enumerable: false,
    configurable: true,
  });
  // https://github.com/tc39/test262/blob/master/test/intl402/ListFormat/constructor/supportedLocalesOf/length.js
  Object.defineProperty(ListFormat.supportedLocalesOf, 'length', {
    value: 1,
    writable: false,
    enumerable: false,
    configurable: true,
  });
} catch (e) {
  // Meta fix so we're test262-compliant, not important
}
