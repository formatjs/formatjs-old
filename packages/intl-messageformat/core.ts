export * from './lib/formatters';
export * from './lib/core';