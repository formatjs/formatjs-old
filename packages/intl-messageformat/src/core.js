/*
Copyright (c) 2014, Yahoo! Inc. All rights reserved.
Copyrights licensed under the New BSD License.
See the accompanying LICENSE file for terms.
*/

/* jslint esnext: true */

import {extend, hop} from './utils';
import {defineProperty, objCreate, fnBind} from './es5';
import Compiler from './compiler';
import parser from 'intl-messageformat-parser';

export default MessageFormat;

// -- MessageFormat --------------------------------------------------------

function MessageFormat(message, locales, formats) {
    // Parse string messages into an AST.
    var ast = typeof message === 'string' ?
            MessageFormat.__parse(message) : message;

    if (!(ast && ast.type === 'messageFormatPattern')) {
        throw new TypeError('A message must be provided as a String or AST.');
    }

    // Creates a new object with the specified `formats` merged with the default
    // formats.
    formats = this._mergeFormats(MessageFormat.FORMATS, formats);

    // Defined first because it's used to build the format pattern.
    defineProperty(this, '_locale',  {value: this._resolveLocale(locales)});

    var pluralFn = MessageFormat.__localeData__[this._locale].pluralFunction;

    // Define the `pattern` property, a compiled pattern that is highly
    // optimized for repeated `format()` invocations. **Note:** This passes the
    // `locales` set provided to the constructor instead of just the resolved
    // locale.
    var pattern = this._compilePattern(ast, locales, formats, pluralFn);
    defineProperty(this, '_pattern', {value: pattern});

    // Bind `format()` method to `this` so it can be passed by reference like
    // the other `Intl` APIs.
    this.format = fnBind.call(this.format, this);
}

// Default format options used as the prototype of the `formats` provided to the
// constructor. These are used when constructing the internal Intl.NumberFormat
// and Intl.DateTimeFormat instances.
defineProperty(MessageFormat, 'FORMATS', {
    enumerable: true,

    value: {
        number: {
            'currency': {
                style: 'currency'
            },

            'percent': {
                style: 'percent'
            }
        },

        date: {
            'short': {
                month: 'numeric',
                day  : 'numeric',
                year : '2-digit'
            },

            'medium': {
                month: 'short',
                day  : 'numeric',
                year : 'numeric'
            },

            'long': {
                month: 'long',
                day  : 'numeric',
                year : 'numeric'
            },

            'full': {
                weekday: 'long',
                month  : 'long',
                day    : 'numeric',
                year   : 'numeric'
            }
        },

        time: {
            'short': {
                hour  : 'numeric',
                minute: 'numeric'
            },

            'medium':  {
                hour  : 'numeric',
                minute: 'numeric',
                second: 'numeric'
            },

            'long': {
                hour        : 'numeric',
                minute      : 'numeric',
                second      : 'numeric',
                timeZoneName: 'short'
            },

            'full': {
                hour        : 'numeric',
                minute      : 'numeric',
                second      : 'numeric',
                timeZoneName: 'short'
            }
        }
    }
});

// Define internal private properties for dealing with locale data.
defineProperty(MessageFormat, '__availableLocales__', {value: []});
defineProperty(MessageFormat, '__localeData__', {value: objCreate(null)});
defineProperty(MessageFormat, '__addLocaleData', {value: function (data) {
    if (!(data && data.locale)) {
        throw new Error('Object passed does not identify itself with a valid language tag');
    }

    if (!data.messageformat) {
        throw new Error('Object passed does not contain locale data for IntlMessageFormat');
    }

    var availableLocales = MessageFormat.__availableLocales__,
        localeData       = MessageFormat.__localeData__;

    // Message format locale data only requires the first part of the tag.
    var locale = data.locale.toLowerCase().split('-')[0];

    availableLocales.push(locale);
    localeData[locale] = data.messageformat;

    // Assign as `defaultLocale` if there isn't already a default.
    if (!MessageFormat.defaultLocale) {
        MessageFormat.defaultLocale = locale;
    }
}});

// Defines `__parse()` static method as an exposed private.
defineProperty(MessageFormat, '__parse', {value: parser.parse});

// Define public `defaultLocale` property which is set when the first bundle of
// locale data is added.
defineProperty(MessageFormat, 'defaultLocale', {
    enumerable: true,
    writable  : true,
    value     : undefined
});

MessageFormat.prototype.format = function (values) {
    return this._format(this._pattern, values);
};

MessageFormat.prototype.resolvedOptions = function () {
    // TODO: Provide anything else?
    return {
        locale: this._locale
    };
};

MessageFormat.prototype._compilePattern = function (ast, locales, formats, pluralFn) {
    var compiler = new Compiler(locales, formats, pluralFn);
    return compiler.compile(ast);
};

MessageFormat.prototype._format = function (pattern, values) {
    var result = '',
        i, len, part, id, value;

    for (i = 0, len = pattern.length; i < len; i += 1) {
        part = pattern[i];

        // Exist early for string parts.
        if (typeof part === 'string') {
            result += part;
            continue;
        }

        id = part.id;

        // Enforce that all required values are provided by the caller.
        if (!(values && hop.call(values, id))) {
            throw new Error('A value must be provided for: ' + id);
        }

        value = values[id];

        // Recursively format plural and select parts' option â€” which can be a
        // nested pattern structure. The choosing of the option to use is
        // abstracted-by and delegated-to the part helper object.
        if (part.options) {
            result += this._format(part.getOption(value), values);
        } else {
            result += part.format(value);
        }
    }

    return result;
};

MessageFormat.prototype._mergeFormats = function (defaults, formats) {
    var mergedFormats = {},
        type, mergedType;

    for (type in defaults) {
        if (!hop.call(defaults, type)) { continue; }

        mergedFormats[type] = mergedType = objCreate(defaults[type]);

        if (formats && hop.call(formats, type)) {
            extend(mergedType, formats[type]);
        }
    }

    return mergedFormats;
};

MessageFormat.prototype._resolveLocale = function (locales) {
    var availableLocales = MessageFormat.__availableLocales__,
        locale, parts, i, len;

    if (availableLocales.length === 0) {
        throw new Error('No locale data has been provided for IntlMessageFormat yet');
    }

    if (typeof locales === 'string') {
        locales = [locales];
    }

    if (locales && locales.length) {
        for (i = 0, len = locales.length; i < len; i += 1) {
            locale = locales[i].toLowerCase().split('-')[0];

            // Make sure the first part of the locale that we care about is
            // structurally valid.
            if (!/[a-z]{2,3}/i.test(locale)) {
                throw new RangeError('"' + locales[i] + '" is not a structurally valid language tag');
            }

            if (availableLocales.indexOf(locale) >= 0) {
                break;
            }
        }
    }

    return locale || MessageFormat.defaultLocale;
};
