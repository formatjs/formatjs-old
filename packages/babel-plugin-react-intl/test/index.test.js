import * as path from 'path';
import * as fs from 'fs';
import * as babel from '@babel/core';
import plugin from '../src';
import uuidv1 from 'uuid/v1';

function trim(str) {
  return str.toString().replace(/^\s+|\s+$/, '');
}

const skipTests = [
  '.babelrc',
  '.DS_Store',
  'enforceDescriptions',
  'extractSourceLocation',
  'moduleSourceName',
  'icuSyntax',
  'removeDescriptions',
  'overrideIdFn',
  'removeDefaultMessage'
];

const fixturesDir = path.join(__dirname, 'fixtures');
const baseDir = path.join(__dirname, '..');

describe('emit asserts for: ', () => {
  fs.readdirSync(fixturesDir).map(caseName => {
    if (skipTests.indexOf(caseName) >= 0) return;

    it(`output match: ${caseName}`, () => {
      const fixtureDir = path.join(fixturesDir, caseName);

      // Ensure messages are deleted
      const actualMessagesPath = path.join(fixtureDir, 'actual.json');
      if (fs.existsSync(actualMessagesPath)) fs.unlinkSync(actualMessagesPath);

      const actual = transform(path.join(fixtureDir, 'actual.js'));

      // Check code output
      
      expect(trim(actual)).toMatchSnapshot();

      // Check message output
      const expectedMessages = fs.readFileSync(
        path.join(fixtureDir, 'expected.json')
      );
      const actualMessages = fs.readFileSync(
        path.join(fixtureDir, 'actual.json')
      );
      expect(trim(actualMessages)).toEqual(trim(expectedMessages));
    });
  });
});

describe('options', () => {
  it('removeDefaultMessage should remove default message', () => {
    const fixtureDir = path.join(fixturesDir, 'removeDefaultMessage');

    const actual = transform(path.join(fixtureDir, 'actual.js'), {
      removeDefaultMessage: true
    });

    // Check code output
    expect(trim(actual)).toMatchSnapshot()

    // Check message output
    const expectedMessages = fs.readFileSync(
      path.join(fixtureDir, 'expected.json')
    );
    const actualMessages = fs.readFileSync(
      path.join(fixtureDir, 'actual.json')
    );
    expect(trim(actualMessages)).toEqual(trim(expectedMessages));
  });
  it('enforces descriptions when enforceDescriptions=true', () => {
    const fixtureDir = path.join(fixturesDir, 'enforceDescriptions');
    expect(() =>
      transform(path.join(fixtureDir, 'actual.js'), {
        enforceDescriptions: true
      })
    ).toThrow(/Message must have a `description`/);
  });

  it('correctly overrides the id when overrideIdFn is provided', () => {
    const fixtureDir = path.join(fixturesDir, 'overrideIdFn');

    const actual = transform(path.join(fixtureDir, 'actual.js'), {
      overrideIdFn: (id, defaultMessage, description) => {
        return `HELLO.${id}.${defaultMessage.length}.${typeof description}`;
      }
    });

    // Check code output
    
    expect(trim(actual)).toMatchSnapshot();

    // Check message output
    const expectedMessages = fs.readFileSync(
      path.join(fixtureDir, 'expected.json')
    );
    const actualMessages = fs.readFileSync(
      path.join(fixtureDir, 'actual.json')
    );
    expect(trim(actualMessages)).toEqual(trim(expectedMessages));
  });

  it('allows no description when enforceDescription=false', () => {
    const fixtureDir = path.join(fixturesDir, 'enforceDescriptions');
    expect(() =>
      transform(path.join(fixtureDir, 'actual.js'), {
        enforceDescriptions: false
      })
    ).not.toThrow();
  });

  it('removes descriptions when plugin is applied more than once', () => {
    const fixtureDir = path.join(fixturesDir, 'removeDescriptions');
    expect(() =>
      transform(
        path.join(fixtureDir, 'actual.js'),
        {
          enforceDescriptions: true
        },
        {
          multiplePasses: true
        }
      )
    ).not.toThrow();
  });

  it('respects moduleSourceName', () => {
    const fixtureDir = path.join(fixturesDir, 'moduleSourceName');
    expect(() =>
      transform(path.join(fixtureDir, 'actual.js'), {
        moduleSourceName: 'react-i18n'
      })
    ).not.toThrow();

    // Check message output
    const expectedMessages = fs.readFileSync(
      path.join(fixtureDir, 'expected.json')
    );
    const actualMessages = fs.readFileSync(
      path.join(fixtureDir, 'actual.json')
    );
    expect(trim(actualMessages)).toEqual(trim(expectedMessages));
  });

  it('respects extractSourceLocation', () => {
    const fixtureDir = path.join(fixturesDir, 'extractSourceLocation');
    expect(() =>
      transform(path.join(fixtureDir, 'actual.js'), {
        extractSourceLocation: true
      })
    ).not.toThrow();

    // Check message output
    const expectedMessages = fs.readFileSync(
      path.join(fixtureDir, 'expected.json')
    );
    const actualMessages = fs.readFileSync(
      path.join(fixtureDir, 'actual.json')
    );
    expect(trim(actualMessages)).toEqual(trim(expectedMessages));
  });
});

describe('errors', () => {
  it('Properly throws parse errors', () => {
    const fixtureDir = path.join(fixturesDir, 'icuSyntax');
    expect(() => transform(path.join(fixtureDir, 'actual.js'))).toThrow(
      /Expected .* but "\." found/
    );
  });
});

const BASE_OPTIONS = {
  messagesDir: baseDir
};

function transform(filePath, options = {}, { multiplePasses = false } = {}) {
  function getPluginConfig() {
    return [
      plugin,
      {
        ...BASE_OPTIONS,
        ...options
      },
      uuidv1()
    ];
  }

  return babel.transformFileSync(filePath, {
    plugins: multiplePasses
      ? [getPluginConfig(), getPluginConfig()]
      : [getPluginConfig()]
  }).code;
}