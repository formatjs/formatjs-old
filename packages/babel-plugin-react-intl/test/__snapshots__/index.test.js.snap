// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emit asserts for:  output match: FormattedHTMLMessage 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedHTMLMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"<h1>Hello World!</h1>\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: FormattedMessage 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"Hello World!\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: defineMessages 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

const msgs = (0, _reactIntl.defineMessages)({
  header: {
    \\"id\\": \\"foo.bar.baz\\",
    \\"defaultMessage\\": \\"Hello World!\\"
  },
  content: {
    \\"id\\": \\"foo.bar.biff\\",
    \\"defaultMessage\\": \\"Hello Nurse!\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.header)), _react.default.createElement(\\"p\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.content)));
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: descriptionsAsObjects 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo.bar.baz\\",
      defaultMessage: \\"Hello World!\\"
    });
  }

}

exports.default = Foo;"
`;

exports[`emit asserts for:  output match: extractFromFormatMessageCallFalse 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class Foo extends _react.Component {
  render() {
    const msgs = {
      baz: this.props.intl.formatMessage({
        id: 'foo.bar.baz',
        defaultMessage: 'Hello World!',
        description: 'The default message'
      }),
      biff: this.props.intl.formatMessage({
        id: 'foo.bar.biff',
        defaultMessage: 'Hello Nurse!',
        description: 'Another message'
      })
    };
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, msgs.header), _react.default.createElement(\\"p\\", null, msgs.content), _react.default.createElement(\\"span\\", null, _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"foo\\",
      defaultMessage: \\"bar\\"
    })));
  }

}

var _default = (0, _reactIntl.injectIntl)(Foo);

exports.default = _default;"
`;

exports[`options correctly overrides the id when overrideIdFn is provided 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

const msgs = (0, _reactIntl.defineMessages)({
  header: {
    \\"id\\": \\"HELLO.foo.bar.baz.12.string\\",
    \\"defaultMessage\\": \\"Hello World!\\"
  },
  content: {
    \\"id\\": \\"HELLO.foo.bar.biff.12.object\\",
    \\"defaultMessage\\": \\"Hello Nurse!\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(\\"div\\", null, _react.default.createElement(\\"h1\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.header)), _react.default.createElement(\\"p\\", null, _react.default.createElement(_reactIntl.FormattedMessage, msgs.content)), _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"HELLO.foo.bar.zoo.12.object\\",
      defaultMessage: \\"Hello World!\\"
    }), _react.default.createElement(_reactIntl.FormattedHTMLMessage, {
      id: \\"HELLO.foo.bar.delta.21.string\\",
      defaultMessage: \\"<h1>Hello World!</h1>\\"
    }));
  }

}

exports.default = Foo;"
`;

exports[`options removeDefaultMessage should remove default message 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require(\\"react\\"));

var _reactIntl = require(\\"react-intl\\");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

const messages = (0, _reactIntl.defineMessages)({
  foo: {
    \\"id\\": \\"greeting-user\\"
  }
});

class Foo extends _react.Component {
  render() {
    return _react.default.createElement(_reactIntl.FormattedMessage, {
      id: \\"greeting-world\\"
    });
  }

}

exports.default = Foo;"
`;
